artemis-server/src/main/java/org/apache/activemq/artemis/spi/core/security/jaas/TextFileCertificateLoginModule.java

```diff
@@ -71,8 +79,12 @@ protected String getUserNameForCertificates(final X509Certificate[] certs) throw
       if (certs == null) {
          throw new LoginException("Client certificates not found. Cannot authenticate.");
       }
-
-      return usersByDn.get(getDistinguishedName(certs));
+      String dn = getDistinguishedName(certs);
+      String name = usersByDn.get(dn);
```

Comment:
Hence this `userByDb` not synchronized access no longer exists in the new commit?
With github and multiple commits is not simple to do reviews :P